test/ctia/test_helpers/aggregate.clj

```diff
@@ -45,25 +46,28 @@
    while (es-get-in {:a [{:b 2} {:b 3}]} [:a :b]) returns '(2 3)"
   {:test (fn []
            (is (= (es-get-in {:a [{:b 2} {:b 3}]} [:a :b])
-                      '(2 3)))
+                  '(2 3)))
            (is (= (es-get-in {:a [2 3]} [:a])
-                      [2 3]))
+                  [2 3]))
            (is (= (es-get-in {:a {:b [2 3]}} [:a :b])
-                      [2 3]))
+                  [2 3]))
            (is (= (es-get-in {:a [{:b 2} {:b 3}]} [:a :b])
-                      '(2 3)))
+                  '(2 3)))
            (is (= (es-get-in {:a {:b 2 :c 3}} [:a :b])
-                      2))
+                  2))
            (is (= (es-get-in {:a {:d 2 :c 3}} [:a :b])
-                      nil))
+                  nil))
            (is (= (es-get-in {:a [{:d 2} {:b 3}]} [:a :b])
-                      '(3))))}
+                  '(3))))}
   [m ks]
   (reduce (fn [acc k]
             (cond
-              (map? acc) (get acc k)
-              (coll? acc) (seq (keep #(get % k) acc))
-              :else (reduced acc)))
+              (map? acc)  (get acc k)
+              (coll? acc) (->> acc
+                               (keep #(es-get-in % [k]))
+                               flatten
```

Comment:
I tend to prefer `apply concat` given:

```
user=> (flatten #{1 2 3})
()
```

Reply:
but `(apply concat #{1 2 3})` is even worse, it throws `IllegalArgumentException`, I think I'm missing something here.

Reply:
A better example:

```clj
dev=> (flatten #{[1] [2] [3]})
()
```

---

There's also the problem that `flatten` is recursive:

```clj
dev=> (apply concat [[1], [2], [[[[[[[[3]]]]]]]]])
(1 2 [[[[[[[3]]]]]]])
dev=> (flatten [[1], [2], [[[[[[[[3]]]]]]]]])
(1 2 3)
```

...99% of the times one wants to flatten just one level. Most of the time`flatten` happens to achieve the same effect (because the coll had depth 1), but if in a future the coll depth changed, one would get undesirable behavior opaquely.

Reply:
yes here we only want to flatten one depth level. I recommend to add the attended behavior to the tests.
BTW, here it becomes complicate, I wanted a somehow generic way to test metrics, but it's really hard to simulate ES behavior. We probably hitting some limits. 
So if it becomes too hard for this PR to use the test helper, feel free to not use it and write more verbose tests.

Reply:
@vemv Damn, you got me to scratch my head really hard :) So, I switched from `flatten` to `apply concat` in this particular case, and it broke the tests, it took me a while to sort it out exactly why. The reason it broke the tests, apparently, sometimes the retrieved collection is already flat, and `flatten` doesn't do anything, see this:

```
(flatten ["foo" "bar"] 
;; => ["foo" "bar"])

(apply concat ["foo" "bar"]) 
;; => (\f \o \o \b \a \r)
```
It seems using `flatten` here is justified.

@ereteog 
> BTW, here it becomes complicate, I wanted a somehow generic way to test metrics, but it's really hard to simulate ES behavior.

I wanted to avoid larger refactors/restructuring at this point. I'm still getting familiar with the problem space and the codebase.